base/v0_5_exp/translate.go

```diff
@@ -217,6 +219,53 @@ func translateLink(from Link, options common.TranslateOptions) (to types.Link, t
 	return
 }
 
+func translateSSHAuthorizedKeyResource(from SSHAuthorizedKeyResource, options common.TranslateOptions) (to types.SSHAuthorizedKey, tm translate.TranslationSet, r report.Report) {
+	tr := translate.NewTranslator("yaml", "json", options)
+
+	if from.Local != nil {
+		c := path.New("yaml", "local")
+
+		if options.FilesDir == "" {
+			r.AddOnError(c, common.ErrNoFilesDir)
+			return
+		}
+
+		// calculate file path within FilesDir and check for
+		// path traversal
+		filePath := filepath.Join(options.FilesDir, *from.Local)
+		if err := baseutil.EnsurePathWithinFilesDir(filePath, options.FilesDir); err != nil {
+			r.AddOnError(c, err)
+			return
+		}
+
+		contents, err := ioutil.ReadFile(filePath)
+		if err != nil {
+			r.AddOnError(c, err)
+			return
+		}
+		sshKey := string(contents)
+		if len(sshKey) == 0 {
+			r.AddOnError(c, errors.New("ssh authorized key must not be empty"))
+			return
+		}
+
+		tm, r = translate.Prefixed(tr, "local", &sshKey, &to)
```

Comment:
This adds `local` to both the source and destination paths, which is not correct.  You'll need to manually set the field and call `AddTranslation` instead.  You can manually check the `TranslationSet` contents with `butane -D`.